### PR TITLE
Finalizing ripple-binary-codec refactor

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Finishing the refactor of ripple-binary-codec

### Context of Change
* Removed `./src/utils` folder, none of the utils were used ( **Note:** also removed `bytes-utils.test.js`, as a result there will be fewer tests)
* Removed dependencies for `inherits` and `lodash`
* Typed and documented functions and classes.
* Linter shows no errors.
* Removed `make-class` from all files.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan
Added no new tests, all tests still pass

## Future Tasks
* Add protocol buffer support
* Compatibility with X-Address format
* Generate `definitions.json` from source code in `rippled`
* Write `README.md`

### CODE REVIEW

File: src/binary.ts [hunk 1]
```diff
@@ -1,36 +1,110 @@
...
+import { STObject } from "./types/st-object";
 
-const { sha512Half, transactionID } = require("./hashes");
+/**
+ * Construct a BinaryParser
+ *
+ * @param bytes hex-string to construct BinaryParser from
+ * @returns A BinaryParser
+ */
+const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes);
```
Comment (keefertaylor) [inline src/binary.ts:18]: why not just use a constructor?

File: src/binary.ts [hunk 2]
```diff
@@ -1,36 +1,110 @@
...
-const makeParser = (bytes) => new BinaryParser(bytes);
-const readJSON = (parser) => parser.readType(coreTypes.STObject).toJSON();
-const binaryToJSON = (bytes) => readJSON(makeParser(bytes));
+/**
+ * Parse BinaryParser into JSON
+ *
+ * @param parser BinaryParser object
+ * @returns JSON for the bytes in the BinaryParser
+ */
+const readJSON = (parser: BinaryParser): Record<string, unknown> =>
```
Comment (keefertaylor) [inline src/binary.ts:26]: can we avoid casting here? 

Also can we type unknown?

File: src/binary.ts [hunk 3]
```diff
@@ -1,36 +1,110 @@
...
-function signingData(tx, prefix = HashPrefix.transactionSig) {
+/**
+ * Serialize an object for signing
+ *
+ * @param tx Transaction to serialize
+ * @param prefix Prefix bytes to put before the serialized object
+ * @returns A Buffer with the serialized object
+ */
+function signingData(
+  tx: Record<string, unknown>,
```
Comment (keefertaylor) [inline src/binary.ts:87]: recommend `transaction` which I think we've been using other places for consistency

File: src/binary.ts (hunk 1 above)
Comment (natenichols) [reply src/binary.ts:18]: Honestly have no clue. If it was me writing it from scratch I wouldn't bother. But this was from the previous iteration of the design, and I didn't want to change too much of what was being exported from files.

I could take it or leave it, I don't think it's a huge deal either way.

File: src/hashes.ts [hunk 4]
```diff
@@ -1,44 +1,76 @@
...
-      return new coreTypes.Hash256(this.finish256());
-    },
-  },
-  undefined
-);
-
-function sha512Half(...args) {
+/**
+ * Class for hashing with SHA512
+ * @extends BytesList So SerializedTypes can write bytes to a Sha512Half
```
Comment (sublimator) [inline src/hashes.ts:8]: Should really implement a bytes sink interface rather than extend from a BytesList

**Pull request merged**